clients/src/main/java/org/apache/kafka/common/protocol/ApiKeys.java

```diff
@@ -98,7 +98,8 @@
     UPDATE_FEATURES(ApiMessageType.UPDATE_FEATURES, false, true),
     ENVELOPE(ApiMessageType.ENVELOPE, true, RecordBatch.MAGIC_VALUE_V0, false, true),
     FETCH_SNAPSHOT(ApiMessageType.FETCH_SNAPSHOT, false, RecordBatch.MAGIC_VALUE_V0, false, true),
-    DESCRIBE_CLUSTER(ApiMessageType.DESCRIBE_CLUSTER);
+    DESCRIBE_CLUSTER(ApiMessageType.DESCRIBE_CLUSTER),
+    DECOMMISSION_BROKER(ApiMessageType.DECOMMISSION_BROKER, false, RecordBatch.MAGIC_VALUE_V0, true, true);
```

Comment:
This is not a controller-only API.

Reply:
Hmm. Isn't decommissioning a broker something only a controller can do? Or maybe I am missing something here.

Reply:
You are right that only the controller can properly handle it, but the broker needs to be able to forward it, so it must also expose it.

Reply:
Ah. Right. I'm curious, then, what would happen if we send a decommission request through the broker being decommissioned? Wouldn't the broker shutdown (as a result of being unregistered) before responding to the AdminClient call?

Reply:
I am not sure. It seems like the decommissioned broker should wait around long enough to give the controller a chance to move replicas elsewhere.